docs/tools/linter.md

```diff
@@ -0,0 +1,122 @@
+## Linter
+
+Deno ships with a built in code linter for JavaScript and TypeScript.
+
+**Note: linter is a new feature and still unstable thus it requires `--unstable`
+flag**
+
+```shell
+# lint all JS/TS files in the current directory and subdirectories
+deno lint --unstable
+# lint specific files
+deno lint --unstable myfile1.ts myfile2.ts
+```
+
+### Available rules
+
+- `ban-ts-comment`
+- `ban-untagged-ignore`
+- `constructor-super`
+- `for-direction`
+- `getter-return`
+- `no-array-constructor`
+- `no-async-promise-executor`
+- `no-case-declarations`
+- `no-class-assign`
+- `no-compare-neg-zero`
+- `no-cond-assign`
+- `no-debugger`
+- `no-delete-var`
+- `no-dupe-args`
+- `no-dupe-keys`
+- `no-duplicate-case`
+- `no-empty-character-class`
+- `no-empty-interface`
+- `no-empty-pattern`
+- `no-empty`
+- `no-ex-assign`
+- `no-explicit-any`
+- `no-func-assign`
+- `no-misused-new`
+- `no-namespace`
+- `no-new-symbol`
+- `no-obj-call`
+- `no-octal`
+- `no-prototype-builtins`
+- `no-regex-spaces`
+- `no-setter-return`
+- `no-this-alias`
+- `no-this-before-super`
+- `no-unsafe-finally`
+- `no-unsafe-negation`
+- `no-with`
+- `prefer-as-const`
+- `prefer-namespace-keyword`
+- `require-yield`
+- `triple-slash-reference`
+- `use-isnan`
+- `valid-typeof`
```

Comment:
Not for now - but it would be cool to have an API to get a list of these rules and their associated descriptions. Then we could do something like `deno lint --rules` to examine them...

Reply:
We don't have rule descriptions yet, but I can whip up `deno lint --rules` pretty quickly. I agree it'd be useful to get list of available rules in terminal. Unfortunately doing so in Clap is a nightmare because of lifetimes: https://github.com/denoland/deno_lint/blob/e2ec70237cb238ee1744bac65feb745f375d7097/examples/dlint/main.rs#L49-L56

On the other note I could add `--all-rules` as well that would apply bigger set of rules. Let me know